sdk/src/main/java/com/google/cloud/dataflow/sdk/options/GcpOptions.java

```diff
@@ -183,24 +184,67 @@ void setCredentialFactoryClass(
   public static class DefaultProjectFactory implements DefaultValueFactory<String> {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProjectFactory.class);
 
+    private static String getUserHome() {
+      return System.getProperty("user.home");
+    }
+
+    private String getActiveConfig() throws IOException {
+      final String defaultConfigName = "default";
+
+      // get active configuration
+      File activeConfFile = new File(
+          getUserHome(),
+          ".config/gcloud/active_config");
```

Comment:
Good question - there is `if` branch for `windows` before, also previous version of the code assumed the same hierarchy for all platforms ([here](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/348/files#diff-50c13c4ad4237983c6a869fe066c523dL196)). What would you suggest?